gen/map.go

```diff
@@ -281,3 +281,147 @@ func (m *mapGenerator) equalsUnhashable(g Generator, spec *compile.MapSpec) (str
 
 	return name, wrapGenerateError(spec.ThriftName(), err)
 }
+
+func (m *mapGenerator) zapMarshaler(
```

Comment:
We should document this one in detail because it has non-obvious behavior.

    // Maps are logged as objects if the key is a string or a typedef of a
    // string. If the key is not a string, maps are logged as arrays of
    // objects with a key and value.
    //
    //   map[string]int32{"foo": 1, "bar": 2}
    //   => {"foo": 1, "bar": 2}
    // 
    //   map[int32]string{1: "foo", 2: "bar"}
    //   => [{"key": 1, "value": "foo"}, {"key": 2, "value": "bar"}]